networkx/algorithms/community/louvain.py

```diff
@@ -219,10 +219,18 @@ def _one_level(G, m, partition, resolution=1, is_directed=False, seed=None):
         out_degrees = dict(G.out_degree(weight="weight"))
         Stot_in = [deg for deg in in_degrees.values()]
         Stot_out = [deg for deg in out_degrees.values()]
+        # Calculate weights for both in and out neighbours
+        nbrs = {}
+        for u in G:
+            nbrs[u] = defaultdict(float)
+            for _, n, wt in G.out_edges(u, data=True):
+                nbrs[u][n] += wt["weight"]
+            for n, _, wt in G.in_edges(u, data=True):
+                nbrs[u][n] += wt["weight"]
```

Comment:
Just a nit, but you can pass the edge attr name directly to the `data` kwarg to make this a little cleaner:

```suggestion
            for _, n, wt in G.out_edges(u, data="weight"):
                nbrs[u][n] += wt
            for n, _, wt in G.in_edges(u, data="weight"):
                nbrs[u][n] += wt
```